src/tools/compiletest/src/runtest.rs

```diff
@@ -2011,6 +2011,17 @@ impl<'test> TestCx<'test> {
 
         match output_file {
             TargetLocation::ThisFile(path) => {
+                // The idea here is to use the env variable to pass the build_base directory to rustc_save_analysis to be used in cases where there is no output directory
+                // Use of double backslash in `base` below is to cater for paths in Windows OS
+                env::set_var(
+                    "RUST_SAVE_ANALYSIS_CONFIG",
+                    format!(
+                        "{{\"output_file\": \"{base}\",\"full_docs\": false,\
+                      \"pub_only\": true,\"reachable_only\": false,\
+                      \"distro_crate\": true,\"signatures\": false,\"borrow_data\": false}}",
+                        base = self.config.build_base.display().to_string().replace("\\", "\\\\"),
```

Comment:
PR failed in CI. This is likely due to the use of `\` in Windows OS paths. The resolution of that is to replace `\` in the path with `\\` as done here.